### PR TITLE
Added .net 8 to test platforms

### PR DESCRIPTION
**Pull request by Romfos**
### 🤔 What's changed?

- .NET sdk was updated to version 8
- Added .NET 8 to test platforms (new test platforms: .net 6, .net 7, .net 8)
- Use c# lang version 12

### ⚡️ What's your motivation? 

.NET 8 soon will be released. This is LTS version and will be good to add it to testing platforms

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
- [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
 - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: dotnet/Directory.Build.props [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <LangVersion>11</LangVersion>
+    <LangVersion>latest</LangVersion>
```
Comment (mpkorstanje) [inline dotnet/Directory.Build.props:4]: What is the benefit of using `latest` over a specific version? I'd rather not see the build break when a new language version is introduced.

**Review by mpkorstanje — CHANGES_REQUESTED** [review]
Looks good overall. But can you rebase on/merge main?

**Comment by Romfos** [comment]
I would like to wait .net 8 release, this is a draft pr, but because you merged it - ok, lets start testing from preview version =)

File: dotnet/Directory.Build.props (hunk 1 above)
Comment (Romfos) [reply dotnet/Directory.Build.props]: I will do it later

**Comment by mpkorstanje** [comment]
My availability is a bit sparse these days. By the time I'll be able to look at this again there is a good chance 8.0 will have been released 😉.

**Pull request merged**